### ISSUE TITLE
Will multi GPU speed up the process of training ?

### ISSUE THREAD

**Issue by MingPap** [AlexeyAB/darknet#2077]
Hi, 
as the question in the title 
hope for answers

**Comment by AlexeyAB** [comment]
@MingPap Hi, 

Yes.

Multi-GPU speedup training: https://github.com/AlexeyAB/darknet#how-to-train-with-multi-gpu
But doesn't speedup detection.

**Comment by MingPap** [comment]
using partially-trained model

> @MingPap Hi,
> 
> Yes.
> 
> Multi-GPU speedup training: https://github.com/AlexeyAB/darknet#how-to-train-with-multi-gpu
> But doesn't speedup detection.

Thanks for reply
actually, I don't understand following tips.

***Train it first on 1 GPU for like 1000 iterations: darknet.exe detector train data/voc.data cfg/yolov3-voc.cfg darknet53.conv.74***

***Then stop and by using partially-trained model /backup/yolov3-voc_1000.weights run training with multigpu (up to 4 GPUs): darknet.exe detector train data/voc.data cfg/yolov3-voc.cfg /backup/yolov3-voc_1000.weights -gpus 0,1,2,3***

why should I stop and use another weight file? 

what will happen if I keep training on multi GPUs by using initial weight and go to the end?

**Comment by MingPap** [comment]
@AlexeyAB    as the last comment

**Comment by AlexeyAB** [comment]
@MingPap 

> Train it first on 1 GPU for like 1000 iterations: darknet.exe detector train data/voc.data cfg/yolov3-voc.cfg darknet53.conv.74

> why should I stop and use another weight file?
> 
> what will happen if I keep training on multi GPUs by using initial weight and go to the end?


This is just recommendation to get stable result.
**The first 1000 iterations is the most unstable part of training.** For this case there are used:

* `burn_in=1000` parameter in cfg-file, so learning rate will be changed by using this formula for the first 1000 iterations: `current_learning_rate = learning_rate_in_cfg * pow (current_iteration/1000, 4)`  https://github.com/AlexeyAB/darknet/blob/49189c22e8c09ba7c11622ec22a72b3678887ed2/src/network.c#L94

* mixed-precision for Tensor Cores are disabled automatically (up to 3000 iterations)

* recommended to train by using only 1 GPU for the first 1000 iterations

----

But you can try to train using multi-gpu from the begining:
`darknet.exe detector train data/voc.data cfg/yolov3-voc.cfg darknet53.conv.74 -gpus 0,1,2,3`

**Comment by MingPap** [comment]
@AlexeyAB  I see. Thank you so much.

**Comment by lvshuaigg** [comment]
@AlexeyAB Hello, I see the above line 94 in your source code is net.power, and you said above is 4, may I ask net.power=4?

**Comment by AlexeyAB** [comment]
@lvshuaigg Yes, `net.power=4` by default: https://github.com/AlexeyAB/darknet/blob/49189c22e8c09ba7c11622ec22a72b3678887ed2/src/parser.c#L650
